### ISSUE TITLE
SBL Link Limits Documentation?

### ISSUE THREAD

**Issue by Hutchy68** [SemanticMediaWiki/SemanticBreadcrumbLinks#40]
### Setup and configuration

- SMW version: 2.5.3
- SBL version: 1.4
- MW version: 1.27.x
- DB (MySQL etc.):

### Question

Noticed that the link creation for forward navigation is limited to 20. 1 shown and 19 in the dropdown. Is there a setting documented somewhere to increase/decrease this?

**Comment by kghbln** [comment]
Currently this appears to be a [hard coded limit.](https://github.com/SemanticMediaWiki/SemanticBreadcrumbLinks/blob/master/src/ByPropertyHierarchicalLinksFinder.php#L184) if I am not mistaken. 20 appears to be quite a wide limit but perhaps this can be turned into a configuration parameter.

**Comment by Hutchy68** [comment]
Thanks @kghbln 

I saw the hard coded limit late last night and after testing moving the limit up and down I see issues. More than 20 and the screen height is limited for the browser window, dropdown is pushed to a dropup and unusable. So I'm thinking 20 was a tested limit hard set to maintain usability.

**Comment by mwjames** [comment]
> So I'm thinking 20 was a tested limit hard set to maintain usability.

Yes, it is a usability concern. There are ways to improve the clarity and display characteristics and I loved to see something like demonstrated below but it is very unlikely for me to find the time to implement this.

https://wiki.opensource.org/bin/Main/Open+Source+Initiative+Working+Groups/Beyond+Licensing+Working+Group+Proposal/#Attachments

![image](https://user-images.githubusercontent.com/1245473/39952927-64061024-5590-11e8-95b0-05dbecfaa56a.png)

**Comment by kghbln** [comment]
>  There are ways to improve the clarity and display characteristics and I loved to see something like demonstrated below but it is very unlikely for me to find the time to implement this.

@mwjames Thanks for your comment and providing a pointer to a nifty implementation solution.

@Hutchy68 Perhaps you would like to poke this?

Is it ok if @kghbln creates a separate task for this enhancement after documenting the limit more visibly?

**Comment by mwjames** [comment]
> Is it ok if @kghbln creates a separate task for this enhancement after documenting the limit more visibly?

+1

**Comment by kghbln** [comment]
> +1

Done. See the references.
